src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -182,6 +184,32 @@ public ControlledResource getControlledResource(
     return wsmResource.castToControlledResource();
   }
 
+  /**
+   * Update the name and description metadata fields of a controlled resource. These are only stored
+   * inside WSM, so this does not require any calls to clouds.
+   *
+   * @param workspaceId workspace of interest
+   * @param resourceId resource to update
+   * @param name name to change - may be null, in which case resource name will not be changed.
+   * @param description description to change - may be null, in which case resource description will
+   *     not be changed.
+   */
+  public void updateControlledResourceMetadata(
+      UUID workspaceId,
+      UUID resourceId,
+      @Nullable String name,
+      @Nullable String description,
+      AuthenticatedUserRequest userReq) {
+    stageService.assertMcWorkspace(workspaceId, "updateControlledResource");
+    validateControlledResourceAndAction(
+        userReq, workspaceId, resourceId, SamControlledResourceActions.EDIT_ACTION);
+    // Name may be null if the user is not updating it in this request.
+    if (name != null) {
+      ValidationUtils.validateResourceName(name);
```

Comment:
If a name exists and I want to clear it, is there a way other than with an empty string?

Reply:
No, you'd have to send an empty string.